dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/integration/RegistryDirectory.java

```diff
@@ -437,71 +426,22 @@ private URL mergeUrl(URL providerUrl) {
     }
 
     private URL overrideWithConfigurator(URL providerUrl) {
-        List<Configurator> localConfigurators = this.configurators; // local reference
-        if (localConfigurators != null && !localConfigurators.isEmpty()) {
-            for (Configurator configurator : localConfigurators) {
-                providerUrl = configurator.configure(providerUrl);
-            }
-        }
-
-        List<Configurator> localAppDynamicConfigurators = consumerConfigurationListener.getConfigurators(); // local reference
-        if (localAppDynamicConfigurators != null && !localAppDynamicConfigurators.isEmpty()) {
-            for (Configurator configurator : localAppDynamicConfigurators) {
-                providerUrl = configurator.configure(providerUrl);
-            }
-        }
-
+        providerUrl = overrideWithConfigurators(this.configurators, providerUrl);

```

Comment:
for the shake of mine understanding would it be possible explain bit here about overrideWithConfigurators process here?

Reply:
done.

Reply:
Would you provide some information here for my understanding purpose ?